WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardView.swift

```diff
@@ -21,7 +22,18 @@ final class BlazeCampaignDashboardViewHostingController: SelfSizingHostingContro
         }
 
         rootView.createCampaignTapped = { [weak self] in
-            self?.navigateToCampaignCreation(source: .myStoreSection)
+            guard let self else { return }
+            if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.blazei3NativeCampaignCreation) {
```

Comment:
This check is duplicated with the one in `navigateToCampaignCreation`. Should we remove it and add the check for `shouldShowProductSelectorView` in  `navigateToCampaignCreation` when the feature flag is on?

Reply:
I tried to implement this on https://github.com/woocommerce/woocommerce-ios/pull/11625/commits/d1f6059d3a12f772bf8937202bb4209fcc1fdd33, with some notes:

1. I also refactored `navigateToCampaignCreation()` to make the conditionals easier to understand. Once we remove web campaign creation hopefully it makes deleting code easier as well.
2. This idea also introduces the need for `fromProductSelector` parameter, since `onProductSelectionStateChanged` calls the `navigateToNativeCampaignCreation` function. Without that parameter, then selecting a product will re-open product selector again. This works but doesn't feel completely right to me. Let me know what you think.